### DISCUSSION TITLE
how to include dll file in exe file to be a one exe file ??

### DISCUSSION THREAD

**Discussion by karadevnet** [jchristn/SuperSimpleTcp#178] [General]
hi mate. i start a one simple project with your cool super simple tcp server it is working fine for now and need to make more options that is easy.

but i try to make it to include dll file in exe file to be a one exe file to all project no succes. i try to some examples of code in my project from stackoverflow but  also not success.

can you suport me alittle to make it include dll file in exe to make a one super simple exe tcp chat client super server/client ???

or how to embed library code to compile to one *.exe  file in output folder ???

any help and variant will be helpfull and will achieve success. :)

![clientserverwork](https://user-images.githubusercontent.com/82239300/216767070-9c0a333b-51e5-44be-a920-93bbba412679.png)

**Comment by jchristn** [comment]
Hi, you may want to include the source in your project, and try something like this.

https://learn.microsoft.com/en-us/dotnet/core/deploying/single-file/overview?tabs=cli

Moving this to a discussion.  It's not an issue with the library.

**Comment by karadevnet** [comment]
hi, first i want to apologies for that i post like a issue. i am not familiar with github so far and now i learn about discussions. best regards for that to you.
i made some pictures for you to see that is my app is NOT A CONSOLE like is shown i help link of microsoft help page i do not have this menu and options in menus that are described in help and also try some other links and menus for this task. no success :(

i try some code examples from stackoverflow to add dll file in exe. i make it happened but i dont know how to load when the app is runing. this is some code need to write to main app form but i don know what a code i need to write. when i add dll file in exe, i notice that the size of exe is with dll inside. using resource system make it to do this, but have not way to read/write/execute it like is outside dll file trough the using directive.
ok, i will post here some pictures about i made and please help me with this task to complete my project. i think is the better way is if you make a simple project to complete this task and send me here or ti my email directly to see what a code i need to write in main for app to run all from single file correctly. i use a vs studio enterprise 2022 if this is helpfully for you.

for console app publish menu is very different from a .net core win 10 visual app. have not this setting in it when i click on publish.
best regards and i wait for your answer

here si some pictures 👍 

![error](https://user-images.githubusercontent.com/82239300/216920382-5779b157-7c18-49f5-9f5e-1de4f0a764b4.jpg)

![error1](https://user-images.githubusercontent.com/82239300/216920390-07fc35f4-d035-481c-8901-cb1890611dda.jpg)

![error2](https://user-images.githubusercontent.com/82239300/216920394-f54afb34-9bc0-4f31-a02d-95073dabd7aa.jpg)

![error3](https://user-images.githubusercontent.com/82239300/216920398-12015028-e95d-4723-af41-623fbd1db90c.jpg)

**Comment by karadevnet** [comment]
ok, new update :) i get the source code and add all *.cs files to a project like a existing files in to a project. i don know how is right, but now out exe file is a single and size is a with size of plus supers simple lib :)

this is the link i found about this warning but i dont know how to change the code to remove warning
https://learn.microsoft.com/en-us/dotnet/csharp/misc/cs0436

but here have a some warning that i get from vs studio and need to know how to make it right. app is working normal like a single file, but i am warn about if i have a need to add more dll and/or cs files will make this warning an error.

Severity	Code	Description	Project	File	Line	Suppression State
Warning	CS0436	The type 'ConnectionEventArgs' in 'D:\C#_CLIENT_SERVER_SOCKET_SYSTEM\Client_Server_Socket_System_TEST_ONE_EXE\Client_Server_Socket_System\ConnectionEventArgs.cs' conflicts with the imported type 'ConnectionEventArgs' in 'SuperSimpleTcp, Version=3.0.7.0, Culture=neutral, PublicKeyToken=null'. Using the type defined in 'D:\C#_CLIENT_SERVER_SOCKET_SYSTEM\Client_Server_Socket_System_TEST_ONE_EXE\Client_Server_Socket_System\ConnectionEventArgs.cs'.	Client_Server_Socket_System	D:\C#_CLIENT_SERVER_SOCKET_SYSTEM\Client_Server_Socket_System_TEST_ONE_EXE\Client_Server_Socket_System\Form1.cs	121	Active


![viber_image_2023-02-06_11-36-34-686](https://user-images.githubusercontent.com/82239300/216937039-0406725c-ef71-46ea-b850-e67d224ee737.jpg)
